File: benchmarks/fixtures/aspnet-multi-hop-path-traversal/Services/DocumentFacade.cs
namespace Secwest.Benchmarks.Services;

public sealed class DocumentFacade
{
    private readonly DocumentService _service;

    public DocumentFacade(DocumentService service)
    {
        _service = service;
    }

    public Task<string> ReadAsync(string path, CancellationToken cancellationToken)
    {
        return _service.ReadAsync(path, cancellationToken);
    }
}

File: benchmarks/fixtures/java-multi-hop-path-traversal/src/main/java/example/DocumentFacade.java
package example;

import java.io.IOException;
import org.springframework.stereotype.Service;

@Service
public final class DocumentFacade {
    private final DocumentService service;

    public DocumentFacade(DocumentService service) {
        this.service = service;
    }

    public String read(String path) throws IOException {
        return service.read(path);
    }
}
